source/common/upstream/upstream_impl.cc

```diff
@@ -1698,7 +1701,14 @@ absl::Status ClusterImplBase::parseDropOverloadConfig(
         "Cluster drop_overloads config denominator setting is invalid : {}. Valid range 0~2.",
         drop_percentage.denominator()));
   }
-  drop_overload_ = UnitFloat(float(drop_percentage.numerator()) / (denominator));
+
+  // If DropOverloadRuntimeKey is not enabled, honor the EDS drop_overload config.
+  // If it is enabled, choose the smaller one between it and the EDS config.
```

Comment:
It looks to me the logic below make more sense:

  If DropOverloadRuntimeKey is not enabled, honor the EDS drop_overload config.
  If DropOverloadRuntimeKey is enabled, then honor the runtime key config.

WDYT?